widgets/tab/tabdemo/tabdemo.go

```diff
@@ -0,0 +1,548 @@
+
+				// Create xLabels map for Memory
+				memXLabels := make(map[int]string)
+				for idx, label := range memTimestamps {
+					memXLabels[idx] = label
+				}
+
+				if err := memLineChart.Series("Memory", memData,
+					linechart.SeriesCellOpts(cell.FgColor(cell.ColorGreen)),
+					linechart.SeriesXLabels(memXLabels),
+				); err != nil {
+					log.Printf("failed to update Memory line chart: %v", err)
+				}
+				memText.Reset()
+				if err := memText.Write(fmt.Sprintf("Memory Usage: %.2f%%", memUsage)); err != nil {
+					log.Printf("failed to write Memory text: %v", err)
+				}
+
+				// Simulate Storage activity
+				storageActivity := rnd.Float64() * 100
+				if len(storageData) >= maxDataPoints {
+					storageData = storageData[1:]
+					storageTimestamps = storageTimestamps[1:]
+				}
+				storageData = append(storageData, storageActivity)
+				storageTimestamps = append(storageTimestamps, currentTime)
+
+				// Create xLabels map for Storage
+				storageXLabels := make(map[int]string)
+				for idx, label := range storageTimestamps {
+					storageXLabels[idx] = label
+				}
+
+				if err := storageLineChart.Series("Storage", storageData,
+					linechart.SeriesCellOpts(cell.FgColor(cell.ColorCyan)),
+					linechart.SeriesXLabels(storageXLabels),
+				); err != nil {
+					log.Printf("failed to update Storage line chart: %v", err)
+				}
+				// Update storage details
+				storageText.Reset()
+				driveNames := []string{"Drive A", "Drive B", "Drive C"}
+				driveCapacities := []float64{500, 1000, 2000} // in GB
+				for i, name := range driveNames {
+					used := rnd.Float64() * driveCapacities[i]
+					percentUsed := (used / driveCapacities[i]) * 100
+					if err := storageText.Write(fmt.Sprintf("%s: Used %.2fGB / %.0fGB (%.2f%%)\n", name, used, driveCapacities[i], percentUsed)); err != nil {
+						log.Printf("failed to write storage text: %v", err)
+					}
+				}
+
+				// Simulate Network traffic
+				networkTraffic := rnd.Float64() * 1000 // Mbps
+				if len(netData) >= maxDataPoints {
+					netData = netData[1:]
+					netTimestamps = netTimestamps[1:]
+				}
+				netData = append(netData, networkTraffic)
+				netTimestamps = append(netTimestamps, currentTime)
+
+				// Create xLabels map for Network
+				netXLabels := make(map[int]string)
+				for idx, label := range netTimestamps {
+					netXLabels[idx] = label
+				}
+
+				if err := networkLineChart.Series("Network", netData,
+					linechart.SeriesCellOpts(cell.FgColor(cell.ColorMagenta)),
+					linechart.SeriesXLabels(netXLabels),
+				); err != nil {
+					log.Printf("failed to update Network line chart: %v", err)
+				}
+				networkText.Reset()
+				if err := networkText.Write(fmt.Sprintf("Network Traffic: %.2f Mbps", networkTraffic)); err != nil {
+					log.Printf("failed to write network text: %v", err)
+				}
+
+				// Sleep before next update
+				time.Sleep(500 * time.Millisecond)
+			}
+		}
+	}()
+
+	// Start a goroutine to periodically trigger tab notifications
```

Comment:
This is the goroutine I am referring to, I would suggest we remove this to make the demo experience more stable.